### ISSUE TITLE
`AwsSession.copy_session()` fails when  "the config profile (default) could not be found"

### ISSUE THREAD

**Issue by cqc-alec** [aws/amazon-braket-sdk-python#453]
**Describe the bug**
When an `AwsSession` is constructed directly from a boto3 session, attempts to copy the session fail

**To reproduce**
```python3
import boto3
from braket.aws.aws_session import AwsSession
boto_session = boto3.Session(
    aws_access_key_id=aws_access_key_id,
    aws_secret_access_key=aws_secret_access_key,
    region_name=region_name,
)
session = AwsSession(boto_session=boto_session)
new_session = session.copy_session(region=new_region)
```

**Expected behavior**
A new session is created.

**Screenshots or logs**
```
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/braket/aws/aws_session.py:795: in copy_session
    boto_session = boto3.Session(
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/boto3/session.py:90: in __init__
    self._setup_loader()
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/boto3/session.py:131: in _setup_loader
    self._loader = self._session.get_component('data_loader')
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/botocore/session.py:796: in get_component
    return self._components.get_component(name)
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/botocore/session.py:1108: in get_component
    self._components[name] = factory()
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/botocore/session.py:193: in <lambda>
    lambda: create_loader(self.get_config_variable('data_path')),
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/botocore/session.py:317: in get_config_variable
    return self.get_component('config_store').get_config_variable(
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/botocore/configprovider.py:426: in get_config_variable
    return provider.provide()
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/botocore/configprovider.py:628: in provide
    value = provider.provide()
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/botocore/configprovider.py:718: in provide
    scoped_config = self._session.get_scoped_config()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <botocore.session.Session object at 0x7fef26ea7a90>

    def get_scoped_config(self):
        """
        Returns the config values from the config file scoped to the current
        profile.
    
        The configuration data is loaded **only** from the config file.
        It does not resolve variables based on different locations
        (e.g. first from the session instance, then from environment
        variables, then from the config file).  If you want this lookup
        behavior, use the ``get_config_variable`` method instead.
    
        Note that this configuration is specific to a single profile (the
        ``profile`` session variable).
    
        If the ``profile`` session variable is set and the profile does
        not exist in the config file, a ``ProfileNotFound`` exception
        will be raised.
    
        :raises: ConfigNotFound, ConfigParseError, ProfileNotFound
        :rtype: dict
    
        """
        profile_name = self.get_config_variable('profile')
        profile_map = self._build_profile_map()
        # If a profile is not explicitly set return the default
        # profile config or an empty config dict if we don't have
        # a default profile.
        if profile_name is None:
            return profile_map.get('default', {})
        elif profile_name not in profile_map:
            # Otherwise if they specified a profile, it has to
            # exist (even if it's the default profile) otherwise
            # we complain.
>           raise ProfileNotFound(profile=profile_name)
E           botocore.exceptions.ProfileNotFound: The config profile (default) could not be found
```

**System information**
- **Amazon Braket Python SDK version**: 1.30.2
- **Amazon Braket Python Schemas version**: 1.10.2
- **Amazon Braket Python Default Simulator version**: 1.9.0
- **Python version**: 3.9.13

**Additional context**
This was not an issue in 1.29.2 and presumably is a consequence of https://github.com/aws/amazon-braket-sdk-python/commit/60fc15b5415c5a628cbd1ba0308f3979e3299260 .

**Comment by ajberdy** [comment]
Thanks for raising this. I will get a fix out asap

**Comment by ajberdy** [comment]
Looks like the issue here is that when no profile is present, `profile_name` will be default. However, initializing a new session with profile name "default" fails without a correspondingly named profile. #454 should resolve this issue.

**Comment by ajberdy** [comment]
Fix is included in the latest release, `1.31.0`. Closing this issue, please feel free to reopen if you run into anymore problems here!

**Comment by cqc-alec** [comment]
@ajberdy Thank you for the quick fix!
